includes/modules/order_total/ot_coupon.php

```diff
@@ -170,17 +170,25 @@ function collect_posts() {
        );
        
       $messageStack->add_session('checkout_payment', TEXT_REMOVE_REDEEM_COUPON, 'caution');
-    }
-//    print_r($_SESSION);
-    // bof: Discount Coupon zoned always validate coupon for payment address changes
-    // eof: Discount Coupon zoned always validate coupon for payment address changes
+    }
+
     if ((isset($_POST['dc_redeem_code']) && $_POST['dc_redeem_code'] != '') || (isset($discount_coupon->fields['coupon_code']) && $discount_coupon->fields['coupon_code'] != '')) {
       // set current Discount Coupon based on current or existing
-      if (isset($_POST['dc_redeem_code']) && $discount_coupon->fields['coupon_code'] == '') {
-        $dc_check = $_POST['dc_redeem_code'];
-      } else {
-        $dc_check = $discount_coupon->fields['coupon_code'];
-      }
+      if ($discount_coupon != null) {
+        if (isset($_POST['dc_redeem_code']) && $discount_coupon->fields['coupon_code'] == '') {

```

Comment:
`$_POST['dc_redeem_code']` is always set, per line 160. Maybe this should be checking for `empty`?

Reply:
Oh wait, maybe not if "removed".